Automattic/wp-calypso
# WPCOM SSH: Allow a user to upload a SSH key to their account

#### Proposed Changes

Closes https://github.com/Automattic/wp-calypso/issues/68201 and gates the SSH CRUD behind the `hosting/ssh-keys` feature flag.

It misses an icon to the left of the key, the date it's been added, and event validation (@danielbachhuber, @vindl). 

Do we really care about the icon and the added date, or is it OK to add them on a follow-up?

#### Testing Instructions

1. Patch your sandbox with D88901-code and make sure you're proxied;
2. Open your profile settings and check that there is a new item in the "Security Checklist" section:

![image](https://user-images.githubusercontent.com/26530524/193914716-167c2deb-cfaa-4922-b6ca-7285449609d1.png)

3. Click on that item, fill the public key text area, and press "Save SSH key":

![image](https://user-images.githubusercontent.com/26530524/193914967-5c6e7610-9a17-42fb-89a2-f61bf06f0502.png)

4. Check the SSH key was added:

![image](https://user-images.githubusercontent.com/26530524/193915040-f508c6b2-93ad-4f40-bd09-420d5933f3ec.png)

5. Click "Remove SSH key" and check that there's confirmation preceding the action:

![image](https://user-images.githubusercontent.com/26530524/193915234-c1f53711-bdac-4a9c-9211-c30603a9b91d.png)

6. After confirming, the SSH key is gone:

![image](https://user-images.githubusercontent.com/26530524/193915334-844c2b4d-728d-4423-a555-ff766b8d81f7.png)

7. Check that the Tracks events were properly dispatched in the success scenarios;
8. Check that the _failure_ Tracks events are also dispatched, including the error code. One quick way to test is to enable the form regardless if there are stored keys (right now, only the "default" one is allowed).
